client/components/course/Revisions/EntitySidebar.vue

```diff
@@ -56,6 +56,11 @@ export default {
 <style lang="scss" scoped>
 $revision-padding: 32px;
 
+.selected-revision {
+  color: #fff !important;
```

Comment:
Can this be done with the `!import` rule?

Reply:
You mean without `!important` rule? I tried, but main selector `.revision` has color on it's own so when I extend this class on `.selected.revision:hover`, it overrides color applied on this selector.

Reply:
The `!important` rule can be avoided in two ways:
1) We use a `mixin` instead of an extended `class`.
2) we continue to use `@extend` on the `.selected-revision` `class` but we move the definition of the `class` below the definition of the `.revision` `class`.

The reason why both would work is the cascading rule of `CSS`. Currently, it does not work because the definition of the `.selected-revision` `class` is above the `.revision`. The way `class` extension works is like this:
 - it takes the names of the `classes` that are extending a `class` and add them to the definition of the rules for the extended `class`. Example:
```
 .a {
   margin: 0;
 }
 
 .b {
   margin: 10px;
 }
 
 .c {
   @extend .a;
   
   color: white;
 }
```
 
 becomes:
 
```
 .a, .c {
   margin: 0;
 }
 
 .b {
   margin: 10px;
 }
 
 .c {
   color: white;
 }
```
 
So, even though in the `SCSS` the definition of the `.c` `class` is below the `.b` `class`. in the final version of the `CSS` the margin rule for the `.c` is above the one for the `.b`. In this case, if there were a tag that had both of these `classes`, the margin rule of the `.b` `class` would be used. The reason why a `mixin` would work is that the `mixins` code is inserted into the `class` definitions or moving the `class` definition below would work is because the extension happens after the `.revision` `class` definition.

tl;dr Use a `mixin` since you are not using the `.selected-revision` `class` in the `HTML`.